### PR TITLE
Get builds working on OSX 

### PR DESCRIPTION
**Pull request by daviddrysdale**
# Checklist

- [ ] Pull request affects core Oak functionality (e.g. runtime, SDK, ABI)
  - [ ] I have written tests that cover the code changes.
  - [ ] I have checked that these tests are run by [Cloudbuild](cloudbuild.yaml)
  - [x] I have updated [documentation](docs/) accordingly.
  - [ ] I have raised an [issue](https://github.com/project-oak/oak/issues) to
        cover any TODOs and/or unfinished work.
- [x] Pull request includes prototype/experimental work that is under
      construction.

### CODE REVIEW

File: oak/proto/log.proto [hunk 1]
```diff
@@ -42,6 +42,6 @@ enum Level {
   ERROR = 1;
   WARN = 2;
   INFO = 3;
-  DEBUG = 4;
+  DEBUGGING = 4;
```
Comment (anghelcovici) [inline oak/proto/log.proto:45]: Why the change?

File: scripts/run_tests [hunk 2]
```diff
@@ -17,9 +17,11 @@ cargo test --all-targets
 cargo test --doc
 cargo clippy --all-targets -- --deny=warnings
 
-bazel_build_flags+=(
-  '--keep_going'
-)
+
+if [[ "${OSTYPE}" == "darwin"*  ]]; then
```
Comment (anghelcovici) [inline scripts/run_tests:21]: What's the * doing? My bash knowledge is failing me

File: sdk/rust/oak/src/stubs.rs [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+//! on the host platform.
+//!
+//! When building the Rust code for an Oak Node, the resulting library code
+//! references the functions provided by the Oak ABI.
+//!
+//! For the target Wasm platform, these Oak ABI functions are provided as Wasm
+//! host functions by the Oak Runtime; however, the Runtime is not available
+//! when compiling on the host platform.
+//!
+//! On some host platforms (e.g. `target_os = "linux"`), linking a library that
```
Comment (anghelcovici) [inline sdk/rust/oak/src/stubs.rs:27]: This is quite annoying :(

File: .bazelrc [hunk 4]
```diff
@@ -9,31 +9,43 @@ build:wasm32 --cpu=wasm32
...
-build:clang --symlink_prefix=bazel-clang-
+build:clang-base --symlink_prefix=bazel-clang-
+
+build:clang --config=clang-base
+build:clang --crosstool_top=//toolchain:clang
+build:clang --host_crosstool_top=@bazel_tools//tools/cpp:toolchain
+build:clang --cpu=k8
+
+# Darwin's default compiler is clang, so no need for a crosstool
+build:darwin --config=clang-base
```
Comment (anghelcovici) [inline .bazelrc:48]: I think the advantage of specifying a "cross" compiler was that it will force a certain version of clang (the one we download) instead of the one available on the system. Not really a big problem, depends how much we want to control that.

File: oak/proto/log.proto (hunk 1 above)
Comment (daviddrysdale) [reply oak/proto/log.proto:45]: "This avoids clashes with C++ builds that use DEBUG as a preprocessor macro." (from the commit message)

File: scripts/run_tests (hunk 2 above)
Comment (daviddrysdale) [reply scripts/run_tests:21]: It matches anything starting with `darwin` (e.g. my `OSTYPE` is `darwin19`).

File: sdk/rust/oak/src/stubs.rs (hunk 3 above)
Comment (daviddrysdale) [reply sdk/rust/oak/src/stubs.rs:27]: Yep.  It took a little bit of figuring out, so I thought a nice big comment would help for the future.

File: .bazelrc (hunk 4 above)
Comment (daviddrysdale) [reply .bazelrc:48]: Having a known version would be nice, but the wrapper scripts in `toolchain/` didn't seem to quite work for a MacOS build, and just falling back to the native toolchain was the easy way out.

I can revisit what was going wrong with `--config=clang` but I might need your help with toolchain wrangling&hellip;

**Comment by daviddrysdale** [comment]
Yeah, I didn't fix that because it was due for deletion in #801 &ndash; and has now been deleted.

File: oak/proto/log.proto (hunk 1 above)
Comment (anghelcovici) [reply oak/proto/log.proto:45]: I can't really see the comments while reviewing, unless I review one by on. Am I just an idiot and didn't notice where they are displayed (when looking at all the changes for the PR)?

File: scripts/run_tests (hunk 2 above)
Comment (anghelcovici) [reply scripts/run_tests:21]: +1 Bash skills
